sotodlib/io/toast_frame_utils.py

```diff
@@ -768,7 +768,7 @@ def split_field(data, g3t, framefield, mapfield=None, g3units=units,
     nnz = 1
     if cache_common_flags is None:
         if rankdet == 0:
-            cflags = tod.read_common_flags(local_start=cacheoff, n=ncache)
+            cflags = np.array(tod.read_common_flags(local_start=cacheoff, n=ncache))
```

Comment:
`read_common_flags` should already by returning an array. Why do you think this is needed?

Reply:
Without this, I have sometimes encountered memory error at clean up process. It may be due to some memory allocation issue between toast and spt3g libraries. Copying into numpy array makes it simple. The flatten() above should be for the same reason.

Reply:
Perhaps @tskisner can take a look at these modifications to the `so3g` interface?

Reply:
Considering that these tools are changing anyway in the toast3 port, I suggest we just merge this before that refactor.